src/renderer/swapchain.rs

```diff
@@ -39,8 +39,11 @@ impl SwapchainWrapper {
         let mut swapchain_create_info = vk::SwapchainCreateInfoKHR::builder()
             .surface(surface.surface)
             .min_image_count(
-                3.max(surface_capabilities.min_image_count)
-                    .min(surface_capabilities.max_image_count),
+                if surface_capabilities.max_image_count > 0 {
```

Comment:
Please make it more readable, maybe using https://doc.rust-lang.org/std/primitive.u32.html#method.clamp and Ord::min(a, b) instead of a.min(b)